### PR TITLE
feat: Add AI-powered PDF invoice parsing

### PR DESCRIPTION
**Pull request by praveenkalady**
#  Add AI-Powered PDF Invoice Parsing

## Overview
This PR adds intelligent PDF parsing to the invoice creation workflow using OpenAI's GPT-4o. When users upload a PDF document, the system automatically extracts invoice data and populates form fields, eliminating manual data entry.

##  Problem Solved
- **Manual data entry**: Users previously had to manually type invoice details from PDF documents
- **Time consumption**: Processing PDF invoices was slow and error-prone
- **Data accuracy**: Manual transcription led to typos and missed information

### Core Features
- **Smart PDF Processing**: Automatically detects and extracts invoice data from uploaded PDFs
- **Seamless Integration**: Uses existing "Add document" button - no UI changes needed
- **Intelligent Field Population**: Auto-fills invoice ID, dates, line items, expenses, and notes
- **Smart Data Conversion**: Handles hourly vs project-based worker quantity conversions
- **Category Matching**: Automatically matches expense categories by name

### User Experience
- **Progress Indicators**: Shows "Parsing PDF with AI..." during processing
- **Error Handling**: Clear, dismissible error messages for invalid PDFs
- **Data Preservation**: Maintains existing form data if parsing fails
- **File Validation**: 10MB size limit with user-friendly error messages

### Dependencies Added
- `@ai-sdk/openai@^2.0.23` - OpenAI integration
- `ai@^5.0.30` - AI SDK core
- Updated `zod@^3.25.76` - Schema validation compatibility

### New Files
- `frontend/app/api/invoices/parse-pdf/route.ts` - OpenAI API integration
- `frontend/app/(dashboard)/invoices/usePdfParsing.ts` - PDF processing hook
- `frontend/models/constants.ts` - PDF size validation constants

### Modified Files
- `frontend/app/(dashboard)/invoices/Edit.tsx` - Integrated parsing into document upload
- `package.json` - Added AI dependencies

### Screen recording

https://github.com/user-attachments/assets/105ac957-9fd3-4ccc-b631-5fe42921bedf

### Test Coverage
- ✅ **6 comprehensive e2e tests** covering all scenarios
- ✅ **Real API integration** (not mocked responses)
- ✅ **Error handling** and edge cases
- ✅ **File validation** and size limits
- ✅ **Form data preservation**
- ✅ **End-to-end workflow**

### Test Results
Screenshot 2025-09-30 at 10 45 02 AM

#### AI disclosure

AI is mostly used for writing test cases for this feature.

### CODE REVIEW

**Comment by praveenkalady** [comment]
@slavingia Can i get a review on this?

File: frontend/app/api/invoices/parse-pdf/route.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    const aiResult = await processWithAI(dataUrl);
+
+    if (!aiResult.success || !aiResult.data) {
+      throw new Error(aiResult.error || "Failed to process PDF with AI");
+    }
+
+    const object = aiResult.data;
+
+    // Check if the result contains any meaningful invoice data
+    const hasData =
```
Comment (praveenkalady) [inline frontend/app/api/invoices/parse-pdf/route.ts:127]: Validation for meaningful invoice data.
An empty response from AI means the PDF wasn't a valid invoice
This prevents users from accidentally importing non-invoice documents

File: frontend/models/constants.ts [hunk 2]
```diff
@@ -210,3 +210,15 @@ export const storageKeys = {
   SETTINGS_MENU_STATE: "settings-menu-state",
   GETTING_STARTED_STATUS: "getting-started-status",
 };
+
+// PDF Upload Configuration
+// Get max file size from environment or use default 10MB
```
Comment (praveenkalady) [inline frontend/models/constants.ts:215]: Constants used for PDF size validation.
